# ciencia-de-proteinas-basada-en-IA

# Tabla de Contenido
- [Introducción](#introducción)
- [Videos recomendados](#Videos-recomendados)
- [Tutoriales](#Tutoriales)
- [Canales de Youtube recomendados](#Canales-de-Youtube-recomendados)
- [ChatBots de ayuda](#ChatBots-de-ayuda)
- [Databases](#ChatBots-de-ayuda)
- [Repositorios utiles](#Repositorios-utiles)
- [Herramientas recomendadas](#Herramientas-recomendadas)
- [Webservers basados en IA](#Webservers-basados-en-IA)
- [Paginas web para aprender sobre las bases de machine learning](#Paginas-web-para-aprender-sobre-las-bases-de-machine-learning)
- [Cursos recomendados](#Cursos-recomendados)
- [Literatura de revision recomendada](#Literatura-de-revision-recomendada)


# introducción
El objetivo de este repositorio es compartir a la comunidad hispanohablante una serie de recursos que seran de utilidad en el aprendizaje de la ciencia de proteinas basada en inteligencia artificial. Para ello se ofrece una serie de tutoriales que pueden ser ejecutados en la nube de Google Colab, una serie de videos (aun por grabar) con clases para comprender la evolucion de las proteinas y las  bases de como funcionan los algoritmos de inteligencia artificial que estan generando una revolucion en el area. Finalmente, se ofrecen un conjunto de herramientas utiles en la practica junto con otros recursos para continuar aprendiendo del tema.  
Nota: texto sin acentos

  

# Videos recomendados
| Tema | Descripcion | Link a Youtube|
|-----------|-----------|-----------|  
|Introduccion a la metagenomica |Reconstruccion de genomas a partir de metagenomas|https://youtu.be/ckIbT93Qhjc| 
|AlphaFold y el Gran Desafío para resolver el plegamiento de proteínas |Breve revision de los algoritmos implementados en AlphaFold y AlphaFold2|https://youtu.be/nGVFbPKrRWQ| 
|Schrödinger y la biología: 75 años del libro ¿Qué es la vida? |Serie de conferencias para comprender el fenomeno d continguencia en evolucion asi como las contribuciones de varios cientificos a la consolidacion de la biologia molecular |Parte 1) https://www.youtube.com/live/XSWqcksA5vg?feature=share&t=711 <br> Parte 2) https://www.youtube.com/live/x35aQO8ifzM?feature=share&t=675 <br> Parte 3) https://www.youtube.com/live/PgbLyOYHEm4?feature=share&t=751| 
|Simbiosis y evolución | Serie de conferencias sobre los niveles de organizacion en biologia con especial enfasis en la simbiosis | Parte 1) https://www.youtube.com/watch?v=dF2GXGcTer8&ab_channel=elcolegionacionalmx <br> Parte 2) https://www.youtube.com/live/PfiZZaa_7BA?feature=share <br> Parte 3) https://www.youtube.com/watch?v=AZT7rf0KNeo&ab_channel=elcolegionacionalmx| 
|La evolución de las proteínas | Charla de introduccion respecto a la evolucion de proteinas|https://youtu.be/HFQqB27Uvbg| 
|Emile Zuckerkandl y el nacimiento de la evolución molecular | Charla de introduccion a la evolucion molecular |[https://youtu.be/ckIbT93Qhjc](https://www.youtube.com/watch?v=qCLgEnSUUmc&ab_channel=elcolegionacionalmx)| 
|Boston Protein Design and Modeling Club | Serie de seminarios sobre los trabajos mas actuales en ingenieria de proteinas |https://www.bpdmc.org/| 
|ML Protein Engineering Seminar Series |Serie de seminarios sobre los trabajos mas actuales en ingenieria de proteinas basado en machine learning |https://www.ml4proteinengineering.com/| 
| Cómo Entender la OCTAVA DIMENSIÓN | Comprender datos de alta dimensionalidad |[https://youtu.be/kAjTrAUFIdQ](https://youtu.be/kAjTrAUFIdQ)| 
| Algoritmo backpropagation | Comprender el algoritmo base de toda IA moderna |[https://www.youtube.com/watch?v=SmZmBKc7Lrs](https://www.youtube.com/watch?v=SmZmBKc7Lrs)| 
| Que es un GPT? | Comprender como funciona la arquitectura transformer |[https://www.youtube.com/watch?v=wjZofJX0v4Ms](https://www.youtube.com/watch?v=wjZofJX0v4M)| 
| Watching Neural Networks Learn | Como funcionan las funciones de activacion |[https://youtu.be/TkwXa7Cvfr8?si=VI38wPQhouvCLZdC](https://youtu.be/TkwXa7Cvfr8?si=VI38wPQhouvCLZdC)| 
| What Is AlphaFold?  | Breve resumen sobre la arquitectura de AlphaFold2 |[https://youtu.be/7q8Uw3rmXyE?si=eAaHDPuZw5Cc7gKB](https://youtu.be/7q8Uw3rmXyE?si=eAaHDPuZw5Cc7gKB)| 
| Origin: Probability of a Single Protein Forming by Chance  | Resumen sobre la evolucion y biofisica de las proteinas |[https://youtu.be/W1_KEVaCyaA?si=kLbw7PH7YqsZK9pG](https://youtu.be/W1_KEVaCyaA?si=kLbw7PH7YqsZK9pG)| 
| Cómo trabajan las enzimas  | Resumen sobre biocatalisis|[https://youtu.be/yk14dOOvwMk?si=SoIA4IogZ9tgmKCb](https://youtu.be/yk14dOOvwMk?si=SoIA4IogZ9tgmKCb)| 
| A History of Bioinformatics (in the Year 2039)  | Resumen sobre los logros y retos de la bioinformatica|[https://www.youtube.com/watch?v=uwsjwMO-TEA&ab_channel=NextDayVideo](https://www.youtube.com/watch?v=uwsjwMO-TEA&ab_channel=NextDayVideo)| 
| Attention is all you need (Transformer) - Model explanation (including math), Inference and Training  | Explicacion paso a paso de la arquitectura transformer |[https://www.youtube.com/watch?v=bCz4OMemCcA&t=925s&ab_channel=UmarJamil](https://www.youtube.com/watch?v=bCz4OMemCcA&t=925s&ab_channel=UmarJamil)| 
| Special Lectures on Machine Learning and Protein Folding | Discusión sobre ls arquitectura de AlphaFold2 |[https://www.youtube.com/playlist?list=PL0NRmB0fnLJQPDZh-6utVnRpF1-beEr4T](https://www.youtube.com/playlist?list=PL0NRmB0fnLJQPDZh-6utVnRpF1-beEr4T)| 
| Review and discussion of AlphaFold3 | Discusión sobre las arquitecturas de AlphaFold |[https://www.youtube.com/watch?v=qjFgthkKxcA](https://www.youtube.com/watch?v=qjFgthkKxcA)| 
| ¿Qué son y cómo se unen los aminoácidos? Al fin entendí: Biología Molecular | introduccion a la formacion de polimeros |[https://www.youtube.com/watch?v=5Ps3QTqo3cM&ab_channel=Alfinentend%C3%AD](https://www.youtube.com/watch?v=5Ps3QTqo3cM&ab_channel=Alfinentend%C3%AD)|
| How do Graphics Cards Work? Exploring GPU Architecture | Como funcionan las GPUs |[https://www.youtube.com/watch?v=h9Z4oGN89MU&ab_channel=BranchEducation](https://www.youtube.com/watch?v=h9Z4oGN89MU&ab_channel=BranchEducation
)| 


# Tutoriales
| Notebook | Descripcion | Link a Google Colab|
|-----------|-----------|-----------|  
| [Analisis_estructural_ESMAtlas](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/analisis_estructural_ESMAtlas.ipynb) | En este cuaderno aprenderas como explorar la base de datos ESMAtlas, descargar los resultados, procesarlos y agrupar estructuras de proteinas | [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/github/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/analisis_estructural_ESMAtlas.ipynb) |
| [sequentially_ESMFold](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/sequentially_ESMFold.ipynb) | En este cuaderno aprenderas como realizar un video del proceso de plegamiento de una proteinas usando ESMFold | [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/github/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/sequentially_ESMFold.ipynb) |
| [Fine tunning of a protein language model. Notebook por Matthew Carrigan - @carrigmat en Twitter](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/protein_language_modeling_by_Matthew_Carrigan.ipynb) | En este cuaderno aprenderas como  re-entrenar (fine-tunning) un lenguaje de proteinas y usarlos para realizar tareas de prediccion a nivel de secuencia y residuo | [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/protein_language_modeling-tf.ipynb) |
| [ESMFold_batch](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/ESMFold_v1_batch.ipynb) | En este cuaderno aprenderas como predecir estructruas de proteinas por lotes usando ESMFold | [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/drive/1C8dGEzR6tHSpcELpftwYjs1EFjLWX9eU#scrollTo=Gw2T5-nHHNO8) |
| [Fine tunning of a protein language model usgin LoRA](https://huggingface.co/blog/AmelieSchreiber/esmbind) | En este tutorial aprenderas a como optimizar los pesos de un modelo de lenguajde de proteinas completo usando la tecnica de optimizacion LoRA a | NAN |
| [ChimeraX](https://dasher.wustl.edu/bio5357/software/chimerax/tutorial.pdf) y [Chimera_X_overview](https://static-bcrf.biochem.wisc.edu/tutorials/chimerax/Chimera_X_overview_document.html) y [más tutoriales especificos](https://www.rbvi.ucsf.edu/chimerax/tutorials.html)| En estos tutoriales aprenderas manipular y visualizar proteinas con ChimeraX | NAN |
| [Genera embeddings](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/generate_esm_embeddings.ipynb) | Genera los embeddings de diferentes modelos de la familia ESM | [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/github/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/generate_esm_embeddings.ipynb) |
| [Descarga proteinas de la AFDB](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/download_afdb_data.ipynb) | En este cuaderno aprenderas como decargar estructruas de proteins y sus datos a partir de la AlphaFold database | [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/github/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/download_afdb_data.ipynb) |
| [Crea un transformer desde 0](https://github.com/naity/protein-transformer) | En repositorioa aprenderas como entrenar un transformer con proteinas | N/A |
| [Parsing_MSA_conservation](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/ESMFold_batch.ipynb) | En este cuaderno aprenderas como añadir informacion de la conservacion de residuos a una estructura .pdb | v1 [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/github/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/notebooks/Parsing_MSA_conservation.ipynb) v2 [![Open In Colab](https://github.com/miangoar/ciencia-de-proteinas-basada-en-IA/blob/main/img/colab_logo.svg)](https://colab.research.google.com/drive/1Og5xK7gwSaMjzykLqJnzCZ0rNGMJMIle?usp=sharing#scrollTo=Oaqb4SPmXbvP) |
| [Easy and accurate protein structure prediction using ColabFold](https://www.nature.com/articles/s41596-024-01060-5) | how to predict mono-, multi-mers and conformations with the ColabFold-AlphaFold2 protocol | N/A |
| [Moving Beyond Memorization](https://github.com/plinder-org/moving_beyond_memorisation) | compare prote structures with plinder | N/A |


https://rbvi.github.io/chimera-tutorials/presentations/modules/chimerax-comp-structures/index.html#/

(see data D2): https://iubmb.onlinelibrary.wiley.com/doi/full/10.1002/bmb.21860

# Canales de Youtube recomendados
| Tema | 
|-----------|
| [CodigoMaquina](https://www.youtube.com/@CodigoMaquina) |
| [Clockworkbio](https://www.youtube.com/@Clockworkbio) |
| [Mutual_Information](https://www.youtube.com/@Mutual_Information) |
| [ArtemKirsanov](https://www.youtube.com/@ArtemKirsanov) |
| [DotCSV](https://www.youtube.com/@DotCSV) |
| [codificandobits](https://www.youtube.com/@codificandobits) |
| [phylumchannel](https://www.youtube.com/@phylumchannel) |


# Data viz
| Tema | 
|-----------|
| [From Data to Viz](https://www.data-to-viz.com/) |
| [The Python Graph Gallery](https://python-graph-gallery.com/) |
| [Python Color Palette Finder](https://python-graph-gallery.com/color-palette-finder/) |
| [dataviz caveats](https://www.data-to-viz.com/caveats.html) |
| [molstar](https://molstar.org/me/) |
| [Biological Sequence Models in the Context of the AI Directives](https://epochai.org/blog/biological-sequence-models-in-the-context-of-the-ai-directives) |
| [Interactive Map of All Pandas Operations](https://blog.dailydoseofds.com/p/interactive-mind-map-of-all-pandas?=) |


# ChatBots de ayuda
| Tema | 
|-----------|
| [ChatGPT](https://chat.openai.com/) |
| [Gemini](https://gemini.google.com/) |
| [claude](https://claude.ai/) |
| [Bing](https://www.bing.com/?cc=es) |
| [HuggingChat](https://huggingface.co/chat/) |
| [biologpt](https://biologpt.com/) |

# Databases
| Repo | Descripcion | 
|-----------|-----------| 
| [uniprot](https://www.uniprot.org/) | annotated proteins |
| [gtdb](https://gtdb.ecogenomic.org/) | microbial genomes |
| [ColabFold Downloads](https://colabfold.mmseqs.com/) | Bases de datos de proteinas (siendo colabfold_envdb_202108 la mas grande, i.e. pesa 110GB)  |
| [Clustering NCBI's nr database](https://github.com/Arcadia-Science/2023-nr-clustering) | clustered nr at 90% length, 90% identity |
| [globdb](https://globdb.org/) | Derreplecated and annotated genomes |
| [protabank](https://www.protabank.org/) | protein-fitness datasets |
| [mavedb](https://www.mavedb.org/) | protein-fitness datasets |
| [revenant](https://revenant.inf.pucp.edu.pe/) | resurrected proteins structures |
| [alphafold DB](https://alphafold.ebi.ac.uk/) | predicted structures for 200M proteins from the uniprot  |
| [esmatlas](https://esmatlas.com/) | predicted structures for 600M proteins from MGnify  |
| [PDB](https://www.rcsb.org/stats/) | 200k protein structures |
| [PDA](https://pragmaticproteindesign.bio.ed.ac.uk/pda/) | ~1400 de novo designed proteins |
| [M-CSA](https://www.ebi.ac.uk/thornton-srv/m-csa/browse/) | Mechanism and Catalytic Site Atlas |
| [huggingface-task](https://huggingface.co/tasks) | a collection of differente ML task and their respective models on HuggingFace |
| [ProteinDesignLabs](https://github.com/Zuricho/ProteinDesignLabs) | List of computational protein design research labs |
| [lukprot](https://zenodo.org/records/13829058) |  eukaryotic predicted proteins based on EukProtDB |



# Repositorios utiles  
| Repo | Descripcion | 
|-----------|-----------| 
| [biolists/folding_tools](https://github.com/biolists/folding_tools) | Listado de herramientas para analisis de proteinas basado en IA |
| [hefeda/design_tools](https://github.com/hefeda/design_tools) | Listado de herramientas para analisis de proteinas basado en IA  |
| [Protein language models](https://github.com/biolists/folding_tools/blob/main/pLM.md) | Listado de modelos de lenguaje de proteinas |
| [biodiffusion](https://github.com/biolists/biodiffusion) | Listado de modelos de difusion especificos para diseño de proteinas  |
| [duerrsimon/folding_tools](https://github.com/duerrsimon/folding_tools/) | Listado de herramientas para analisis de proteinas basado en IA  |
| [AlphaFold2 mini](https://twitter.com/sokrypton/status/1535857255647690753) | Una version optimizada de AF2 para correr en segundos las predicciones a costa de perder un poco de calidad (se recomienda usarlo solo para fines de aprendizaje del plegamiento de las proteinas) |
| [Paper for protein desing with deep learning ](https://github.com/Peldom/papers_for_protein_design_using_DL) | Una lista curada de todos los trabajos de diseño de proteinas basado en aprendisaje profundo  |
| [ColabFold](https://github.com/sokrypton/ColabFold) | Implementacion de AF2 en la nube de Google colab  |
| [Articulos curados por categoria](https://github.com/Peldom/papers_for_protein_design_using_DL) | Una lista de articulos de ciencia de proteinas basada en IA  |
| [biomodes](https://abeebyekeen.com/biomodes-biomolecular-design/) | Una lista de Modelos de IA curados por categoria  |
| [List of papers about Proteins Design using Deep Learning](https://github.com/Peldom/papers_for_protein_design_using_DL) | Literatura curada |
| [PDB documentation](https://www.rcsb.org/docs/general-help/organization-of-3d-structures-in-the-protein-data-bank) | Descripcion de todo lo relacionado a la PDB y como se analizan las proteinas |
| [AFDB foldseek clusters](https://afdb-cluster.steineggerlab.workers.dev) | Datos sobre los ~2.3M clusters de estructuras |

/

# Herramientas recomendadas
| Repo | Descripcion | 
|-----------|-----------| 
| [SeqKit](https://bioinf.shenwei.me/seqkit/) & [short tutorial](https://sandbox.bio/tutorials/seqkit-intro)| Manipulacion de secuencias genomicas |
| [hh-suite](https://bioinf.shenwei.me/seqkit/) | Busqueda de secuencias homólogas remotas  |
| [Diamond2](https://github.com/bbuchfink/diamond) | Blasteo de secuencias de proteinas a escala masiva |
| [MMseqs2](https://github.com/soedinglab/MMseqs2) | Blasteo de secuencias de proteinas a escala masiva |
| [FoldSeek](https://github.com/steineggerlab/foldseek) | Blasteo estructural de proteinas a escala masiva|
| [ProtLearn ](https://github.com/tadorfer/protlearn) | Codificacion de secuencias de proteinas en vectores para el entrenamiento de algoritmos de machine learning |
| [Pfeature](https://github.com/raghavagps/Pfeature) |Codificacion de secuencias de proteinas en vectores para el entrenamiento de algoritmos de machine learning  |
| [Graphein](https://github.com/a-r-j/graphein) | Codificacion de estructuras de proteinas en vectores usando  teoria de grafos |
| [PyUUL](https://pyuul.readthedocs.io/index.html) | Codificacion de estructuras de proteinas en vectores usando algoritmos de analisis 3D |
| [bio_embeddings](https://github.com/sacdallago/bio_embeddings) | Implementacion de varios lenguajes de proteinas  |
| [TRILL ](https://github.com/martinez-zacharya/TRILL) | Implementacion de varios lenguajes de proteinas  |
| [Graph-Part](https://github.com/graph-part/graph-part) |  Preparacion de datasets de secuencias de proteinas  |
| [LazyPredict](https://github.com/shankarpandala/lazypredict) | Comparacion automatica de varios algoritmos de clasificacion y regresion |
| [MolecularNodes](https://github.com/BradyAJohnston/MolecularNodes) | Visualizacion de proteinas con calidad profesional usando Blender |
| [ProDy](http://prody.csb.pitt.edu/tutorials/) | Suite de herramientas de analisis de proteinas |
| [PyPDB](https://github.com/williamgilpin/pypdb) | API para interactuar con la PDB con python |
| [LocalPDB](https://github.com/labstructbioinf/localpdb) | API para interactuar con la PDB con python |
| [pdb-tools](https://github.com/haddocking/pdb-tools) | Herramientas para procesamiento y analisis de archivos .pdb |
| [seqlike](https://github.com/modernatx/seqlike) | Manipulacion y representacion de secuencias de proteinas |
| [BioNumpy](https://github.com/bionumpy/bionumpy/) | Manipulacion y representacion de secuencias usando numpy |
| [pdbfixer](https://github.com/openmm/pdbfixer) | Correccion de errores en archivos .pdb|
| [ProteinFlow](https://github.com/adaptyvbio/ProteinFlow) | procesamiento de secuencias y estructruas para tareas de machine learning|
| [cath DB](https://www.cathdb.info/version/v4_3_0/superfamily/3.40.710.10) | Base de datos de plegamientos proteicos. Ejemplo con la superfamilia de serinbetalactamasas|
| [proteinshake](https://github.com/BorgwardtLab/proteinshake) | procesamiento de estructruas para tareas de machine learning|
| [EvoProtGrad](https://github.com/NREL/EvoProtGrad) | Simular experimentos de evolucion dirigida usando modelos de lenguaje de proteinas|
| [ConservFold](https://www.rodrigueslab.com/resources) | Mapear los residuos conervados a una estructrua usando AlphaFold2 |
| [docktgrid](https://github.com/gmmsb-lncc/docktgrid) | Codificacion de estructuras de proteinas y sus ligandos en vectores usando algoritmos de analisis 3D |
| [pdbeccdutils](https://pdbeurope.github.io/ccdutils/index.html) y [PDBe Arpeggio](https://github.com/PDBeurope/arpeggio) | Manipulacion de pequeñas moluclas de la PDB asi como su interaccion con proteinas |
| [PDBminer](https://github.com/ELELAB/PDBminer) | Manipulacion modelos de la PDB y AlphaFold Database  |
| [pdbe-api-training](https://github.com/glevans/pdbe-api-training) | Tutoriales para aprender a usar la ePDB  |
| [Prop3D](https://github.com/bouralab/Prop3D) | Herramientas para la preparacion de datasets de proteinas para machine learning  |
| [ProteinMPNN in JAX](https://github.com/sokrypton/ColabDesign/tree/main/mpnn) | Una implementacion optimizada de ProteinMPNN en JAX  |
| [BioPandas](https://biopandas.github.io/biopandas/) | Una implementacion de pandas para manipular archivos .pdb  |
| [USalign](https://github.com/pylelab/USalign) | Alineamiento de estructruas monomericas y heterodimeros o en combinacion con acidos nucleicos  |
| [protkit](https://github.com/silicogenesis/protkit) | Manipulacion y representacion de proteinas |
| [foldcomp](https://github.com/steineggerlab/foldcomp) | Algoritmo de compresion de estructruas de proteinas y bases de datos masivas |
| [cath-tools](https://github.com/UCLOrengoGroup/cath-tools) |  Manipulacion y calculos con estructruas proteicas |
| [Pyrcsbsearchapi](https://github.com/rcsb/py-rcsbsearchapi) |  API de la PDB |
| [protestar](https://github.com/refresh-bio/protestar) | Algoritmo de compresion de estructruas de proteinas |
| [consurf](https://consurf.tau.ac.il/consurf_index.php) | Identification de regiones conervadas de proteinas |
| [colav](https://github.com/Hekstra-Lab/colav) | analisis de ensambles conformacionales |
| [AncFlow](https://github.com/rrouz/AncFlow) | pipeline para reconstruccion ancestral |
| [Merizo-search](https://github.com/psipred/merizo_search) | domain segmentation |
| [Foldmason](https://github.com/steineggerlab/foldmason) | Multiple Protein Structure Alignment at Scale |
| [AlphaPulldown](https://www.embl-hamburg.de/AlphaPulldown/) | Complex moedeling with AF-Multimer |
| [ColabDock](https://github.com/JeffSHF/) | Docking protein-proteina |
| [ColabDesign](https://github.com/sokrypton/ColabDesign) | inverse ffolding |
| [LazyAF](https://github.com/ThomasCMcLean/LazyAF) | protein-protein interaction with AF2|
| [progres](https://github.com/greener-group/progres) | Fast protein structure searching by structural embeddings |
| [pLM-BLAST](https://github.com/labstructbioinf/pLM-BLAST) | Remote homology by PLM |
| [CombFold](https://github.com/dina-lab3D/CombFold) | structure of large protein complexes |
| [ChimeraX apps](https://cxtoolshed.rbvi.ucsf.edu/apps/all) | chimeraX  extensions |
| [masif](https://github.com/LPDI-EPFL/masif) | protein desing based on functionl surface patches |
| [Cfold](https://github.com/patrickbryant1/Cfold) | Structure prediction of alternative protein conformations |
| ["Alanin-scanning" with Alphafold2](https://colab.research.google.com/github/sokrypton/ColabBio/blob/main/notebooks/replacement_scan.ipynb) | Structure prediction with a fraction of the aa replaced with Ala [Twitter](https://x.com/sokrypton/status/1812769477228200086) |
| [ColabProTrek](https://colab.research.google.com/drive/1On2xQU0d7351bIBgZpz2T0VUp2gZium0?usp=sharing) and [huggingface server](https://huggingface.co/spaces/westlake-repl/Demo_ProTrek_650M_UniRef50)| Multimodal protein annotation |
| [MolPipeline](https://github.com/basf/MolPipeline) | processing molecules with RDKit in scikit-learn |
| [Prop3D](https://github.com/bouralab/Prop3D) | toolkit for protein structure dataset creation and processing |
| [Colab forms](https://colab.research.google.com/notebooks/forms.ipynb) | convert a colab notebook to an UI |
| [freesasa](https://github.com/mittinatten/freesasa) | for calculating Solvent Accessible Surface Areas |
| [protein_scoring](https://github.com/seanrjohnson/protein_scoring) | Generating and scoring novel enzyme sequences with a variety of models and metrics |
| [roshambo](https://github.com/molecularinformatics/roshambo) | molecular shape comparison |
| [openstructure](https://openstructure.org/docs/2.8/) | protein structure , complexes and docking comparison |
| [AF_unmasked](https://github.com/clami66/AF_unmasked) | structure prediction for huge protein complexes (up to ~27 chains and ~8400 amino acids) |
| [opendock](https://github.com/guyuehuo/opendock)| Protein-Ligand Docking and Modeling |
| [pyScoMotif](https://github.com/3BioCompBio/pyScoMotif) | protein motif search |
| [folddisco](https://github.com/steineggerlab/folddisco) | protein motif search |
| [FlexMol](https://github.com/Steven51516/FlexMol) | construction and evaluation of diverse model architectures |
| [pyRMSD](https://github.com/salilab/pyRMSD) | RMSD calculations of large sets of structures |
| [PLMSearch](https://github.com/maovshao/PLMSearch) | accurate and fast homologous protein search with PLMs |
| [iSeq](https://github.com/BioOmics/iSeq) | Download (meta)data from GSA, SRA, ENA, and DDBJ |
| [xxx](xxx) | xxx |

# Webservers basados en IA  
| Web | Descripcion | 
|-----------|-----------| 
| [ProteInfer](https://google-research.github.io/proteinfer/) | predicting the functional properties of a protein from its amino acid sequence using neural networks |
| [GoPredSim](https://embed.protein.properties/) | Predict protein properties from embeddings |
| [ProteinMPNN](https://huggingface.co/spaces/simonduerr/ProteinMPNN) | Fixed backbone design |
| [DeepFRI](https://beta.deepfri.flatironinstitute.org/) | structure-based protein function prediction (and functional residue identification) method using Graph Convolutional Networks with Language Model features |
| [Amber relaxation](https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/beta/relax_amber.ipynb) | Usa AMBER para relajar tus estructruas de proteinas y evitar clashes estericos|
| [Model Memory Calculator](https://huggingface.co/spaces/hf-accelerate/model-memory-usage) | Calcula el peso de cualquier modelo almacenado en HuggingFace|
| [alphafind](https://alphafind.fi.muni.cz/search) | busqueda de estructruas en la AlphaFold database|
| [af2bind](https://colab.research.google.com/github/sokrypton/af2bind/blob/main/af2bind.ipynb) | Prediccion de sitios de union a sustratos pequeños |
| [DiffDock-Web](https://huggingface.co/spaces/reginabarzilaygroup/DiffDock-Web) | Docking de proteinas con ligandos |
| [ligandMPNN](https://colab.research.google.com/github/ullahsamee/ligandMPNN_Colab/blob/main/LigandMPNN_Colab.ipynb) | Notebook para usar LigandMPNN en google colab|
| [ESMFold](https://esmatlas.com/resources?action=fold) | Weberver para plegar proteins con ESMFold |
| [Foldseek clusters](https://cluster.foldseek.com/) | Weberver para identificar clusters de estructuras en la AFDB |
| [damietta](https://damietta.de/) | protein design toolkit |
| [Model Memory Calculator](https://huggingface.co/spaces/hf-accelerate/model-memory-usage) and [gpu_poor calculator](https://github.com/RahulSChand/gpu_poor)| calculate how much vRAM is needed to train and perform inference on a model hosted Hugging Face |
| [AlphaFold Clusters](https://cluster.foldseek.com/) | investigate the Foldseek clustered AlphaFold database |
| [protrek server](http://search-protrek.com/) | search proteins by seq, structure and funtional annotations |
| [easifa](http://easifa.iddd.group/) | active and binding site annotations for enzymes |



# Paginas web para aprender sobre las bases de machine learning   
| Web | Descripcion | 
|-----------|-----------| 
| [MLU-EXPLAIN](https://mlu-explain.github.io/) | Blogs interactivos de conceptos y algortimos base de machine learning |
| [Seeing-Theory](https://seeing-theory.brown.edu/) | Una introduccion a probabilidad y estadistica con animaciones didacticas  |
| [Distill publications](https://distill.pub/) | Blogs interactivos sobre algoritmos de machine y deep learning |
| [Neural Network SandBox](https://playground.tensorflow.org/#activation=tanh&batchSize=10&dataset=circle&regDataset=reg-plane&learningRate=0.03&regularizationRate=0&noise=0&networkShape=4,2&seed=0.05854&showTestData=false&discretize=false&percTrainData=50&x=true&y=true&xTimesY=false&xSquared=false&ySquared=false&cosX=false&sinX=false&cosY=false&sinY=false&collectStats=false&problem=classification&initZero=false&hideText=false) | Blog para comprender las bases del funcionamiento de las redes neuronales |
| [Stats illustrations](https://github.com/allisonhorst/stats-illustrations) | Ilustraciones para comprender conecptos base de estadistica |
| [3Blue1Brown ](https://www.youtube.com/@3blue1brown) | Videos con interpretaciones graficas sobre conceptos matematicos |
| [cpu.land](https://cpu.land/) | Una breve introduccion al computo en CPUs |
| [tensorflow SandBox](https://playground.tensorflow.org/) | Pagina interactiva para aprender las bases del funcionamiento de las redes neuronales|
| [generative-ai](https://ig.ft.com/generative-ai/) | Introduccion a modelos generativos|
| [diffusion](https://andrewkchan.dev/posts/diffusion.html) | Introduccion a modelos de difusion|
| [The Illustrated AlphaFold](https://elanapearl.github.io/blog/2024/the-illustrated-alphafold/) | explicacion de la arquitectura de AlphaFold3 |
| [Transformer Explainer: Interactive Learning of Text-Generative Models](https://huggingface.co/papers/2408.04619) | version interactiva de la arquitectura de GPT2 |
| [LLM Visualization](https://bbycroft.net/llm) | version interactiva de la arquitectura base de GPT |



# Cursos recomendados
| Tema | Descripcion | 
|-----------|-----------| 
| [Algoritmos en Bioinformática Estructural](https://eead-csic-compbio.github.io/bioinformatica_estructural/) | Biologia estructural|
| [Cloud-based Tutorials on Structural Bioinformatics](https://github.com/pb3lab/ibm3202) | Biologia estructural |
| [Bash/Linux](https://vinuesa.github.io/intro2linux/index.html) | Introduccion al interprete de linux, Bash |
| [LLM](https://txt.cohere.com/llm-university/) | Introduccion a los grandes modelos de lenguaje (Large Language Models; LLM) |
| [Guide to undestanding PDB data](https://www.rcsb.org/news/feature/646e671b1d621d75127a7a52) | Una compilacion de informacion capturada por los creadores de la PDB para aprender sobre el formato de los datos |
| [regexlearn](https://regexlearn.com/es) | Breve curso para aprender el uso de expresiones regulares |
| [BLAST](https://vinuesa.github.io/TIB-filoinfo/sesion3_BLAST/) | Breve curso para aprender BLAST|
| [IA generativa](https://microsoft.github.io/generative-ai-for-beginners/#//) | Breve curso de microsoft para aprender sobre modelos generativos |
| [Alice’s Adventures in a differentiable wonderlandPermalink](https://www.sscardapane.it/alice-book) | Introduccion al algebra y calculo usados en deep learning y sus implementaciones actuales |
| [pandas-cookbook](https://github.com/jvns/pandas-cookbook/) | Breve curso para aprender a manejar datos con Pandas en python|
| [sandbox BIO](https://sandbox.bio/) | Breves tutoriales de las herramientas mas comunes en bioinformatica|
| [glittr](https://glittr.org/) | lista curada de varios cursos de bioinformatica |
| [Estefanía Tarifeño](https://www.youtube.com/@prof.estefaniatarifeno1108/playlists) | Cursos de introduccion a bioinformatica y proteinas |
| [The Matrix Calculus You Need For Deep Learning](https://explained.ai/matrix-calculus/) | tal cual hermano |
| [Build a Large Language Model (From Scratch)](https://github.com/rasbt/LLMs-from-scratch) | tal cual hermano |
| [3Blue1Brown - Algebra lineal](https://www.youtube.com/watch?v=fNk_zzaMoSs&list=PLZHQObOWTQDPD3MizzM2xVFitgF8hE_ab) | tal cual hermano |
| [3Blue1Brown - Algebra lineal (En español)](https://www.youtube.com/@3blue1brownespanol/playlists) | tal cual hermano |
| [Protein Modeling and Design with PyRosetta and Machine Learning](https://www.youtube.com/playlist?list=PLFavr8uo6kSqDB1cbXUDFZSooNk537teC) | tal cual hermano |
| [PyRosetta](https://rosettacommons.github.io/PyRosetta.notebooks/), [Youtube playlist](https://www.youtube.com/playlist?list=PLFavr8uo6kSqDB1cbXUDFZSooNk537teC) and the [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8813091/)| tal cual hermano |
| [ml-engineering](https://github.com/stas00/ml-engineering) | tal cual hermano |
| [How to Create a Protein](https://github.com/universvm/how_to_create_a_protein) | tal cual hermano |
| [rosettaworkshop 2024](https://meilerlab.org/rosetta-workshop-2024/) | tal cual hermano |
| [Applied Machine Learning in Python: a Hands-on Guide with Code](https://geostatsguy.github.io/MachineLearningDemos_Book/intro.html) | tal cual hermano |
| [Large Language Model Course](https://github.com/mlabonne/llm-course) | tal cual hermano |
| [AlphaFold Decoded](https://www.youtube.com/watch?v=NSvp7RFegEs&list=PLJ0WcPQS7xJVJr6ceIPFSkAGAgrkmw1c9&ab_channel=AlphaFoldDecoded) and the [iGEM proyect](https://2024.igem.wiki/aachen/education)| tal cual hermano |
| [Python fundamentals](https://www.youtube.com/playlist?list=PLp0BA-8NZ4bheYbcihGJLg60YSgQRWOcL) | tal cual hermano |
| [Introduction to Python](https://carpentries-incubator.github.io/pangenomics-python/) | tal cual hermano |
| [Generalidades Linux](https://bioinformatica.ciad.mx/generalidades_linux) | tal cual hermano |
| [A-microbiologist-wants-to-compute](https://github.com/SvetlanaUP/A-microbiologist-wants-to-compute) | tal cual hermano |
| [xxx](xxxx) | tal cual hermano |


# Literatura de revision recomendada
| Tema | Descripcion | Link a Youtube|
|-----------|-----------|-----------|  
|Celebrating 50 Years of Journal of Molecular Evolution | Serie de pequeñas revisiones sobre evolucion molecular, su modelado e implementaciones  |https://link.springer.com/journal/239/topicalCollection/AC_ad1951b211df6035aed9ade2172865c4| 
|Scientific Large Language Models: A Survey on Biological & Chemical Domains | Aplicacion de LLM en las ciencias naturales |https://arxiv.org/abs/2401.14656| 
|Loss Functions and Metrics in Deep Learning | Listado de multiples loss functions |https://arxiv.org/abs/2307.02694| 
|An Introduction to Sequence Similarity (“Homology”) Searching | Introduccion al algortimo de BLASTp |https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3820096/| 

## Agradecimientos 
- A [Lorenzo Segovia](https://www.ibt.unam.mx/perfil/3432/dr-lorenzo-patrick-segovia-forcella), [Alejandro Garciarrubio](https://www.ibt.unam.mx/perfil/1956/dr-alejandro-angel-garciarubio-granados) y [Jose A. Farias](https://www.fariaslab.org/)
- A [Xaira Rivera](https://github.com/xairigu) y Alejandro Alarcon 
