# Learning resources
- [Protein science](#protein-science)
- [Bioinformatics](#bioinformatics)
- [Machine Learning, Deep Learning, and Data Science](#machine-learning-deep-learning-and-data-science)
- [Transformers, LLMs, Diffusion Models and Generative AI](#transformers-llms-diffusion-models-and-generative-ai)
- [Python and Computer Science](#python-and-computer-science)
- [Math](#math)
- [Data visualization](#data-visualization)
- [Blogs](#blogs)

Note: resources marked with ⭐ are highly recommended

# Protein science
## Courses
|  | Name | Description |
|-----------|-----------|-----------|
| ⭐ | [DL4Proteins Workshops](https://github.com/Graylab/DL4Proteins-notebooks) | Notebooks covering deep learning tools for biomolecular structure prediction and design |
| | [AlphaFold 3 Overview](https://alphafoldserver.com/guides) | Understanding confidence scores, strengths, limitations, and basic theory about AlphaFold3 |
| | [AlphaFold. A practical guide](https://www.ebi.ac.uk/training/online/courses/alphafold/) | Understanding confidence scores, strengths, limitations, and basic theory about AlphaFold2  | 
| ⭐ | [AlphaFold Decoded](https://2024.igem.wiki/aachen/education ) | An explanation of the AlphaFold2 architecture, starting from the basics of tensors, deep learning, and geometry (Check out the [Youtube channel](youtube.com/@KilianMandon)) | 
| | [AlphaFold Education Summit](https://www.ebi.ac.uk/training/materials/alphafold-education-summit-materials/)| Learn how to integrate AlphaFold into scientific research and education |
| | [Rosetta Workshop 2024](https://meilerlab.org/rosetta-workshop-2024/) | Protein design with Rosetta and associated tools (check out the [2023 version](https://meilerlab.org/rosetta-workshop-2023/)) |
| | [How to Create a Protein](https://github.com/universvm/how_to_create_a_protein) | Basics of biology and protein design for early-career students | 
| | [Protein Modeling and Design with PyRosetta and Machine Learning](https://rosettacommons.github.io/PyRosetta.notebooks/) | Introduction to the fundmamental principles and tools for using PyRosetta (Check out the [Youtube playlist](https://www.youtube.com/playlist?list=PLFavr8uo6kSqDB1cbXUDFZSooNk537teC) and the [Reference paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8813091/)) |
| | [Estefanía Tarifeño](https://www.youtube.com/@prof.estefaniatarifeno1108/playlists) | Introduction to protein science and bioinformatics (in Spanish) |
| | [Cloud-based Tutorials on Structural Bioinformatics](https://github.com/pb3lab/ibm3202) | 12 Tutorials on protein folding, function, structure, dynamics and evolution using Google Colab (Check out the [reference paper](https://pubs.acs.org/doi/10.1021/acs.jchemed.1c00022)) |
| | [Algoritmos en Bioinformática Estructural](https://eead-csic-compbio.github.io/bioinformatica_estructural/) | Biochemical fundamentals and algorithms used in protein science (in Spanish) |
| ⭐ | [Structural Bioinformatics Course 2023](https://structural-bioinformatics.netlify.app/)|  Structural bioinformatics and tools you can use it to visualize, analyze and design proteins |
| | [Modeling proteins using AlphaFold and ColabFold](https://docs.google.com/document/d/1_g1_M-I40CqOQc5obwAt08YntC5D2Z_WNz6mYuUQtyc/edit?tab=t.0#heading=h.m7ei2f72v2ig)| How to predict protein structures using ColabFold and analyze them with ChimeraX |
| | [Introduction to Molecular Animation with ChimeraX](https://pdb101.rcsb.org/train/training-events/animation) | How to use ChimeraX to create animations | 
| |[Practical Simulations for Molecules and Materials](https://www.erastova.xyz/teaching/practical-simulations-for-molecules-and-materials/)| molecular modeling |
| |[DeepChem Step-by-Step Tutorial](https://deepchem.io/tutorials/the-basic-tools-of-the-deep-life-sciences/)| Modeling Proteins with deep learning, Protein ligand modeling, Bioinformatics, Molecular ML and more with DeepChem|
| ⭐ |[Teaching protein structure and function through molecular visualization](https://iubmb.onlinelibrary.wiley.com/doi/full/10.1002/bmb.21860)| tutorials that introduce students to molecular graphics with ChimeraX (Download the Supporting Information)|
| | [Moving Beyond Memorization](https://github.com/plinder-org/moving_beyond_memorisation) | training and evaluating deep learning methods to predict protein-protein and protein-ligand complex |

## Useful publications
|  | Name | Description |
|-----------|-----------|-----------|
| | [PDBx/mmCIF Dictionary Resources](https://mmcif.wwpdb.org/)| Information about the format, dictionaries and tools used by the PDB to define entries | 
| | [alphafold2 explained](https://www.ai4pharm.info/alphafold2)| An explanation of the architecture of alphafold2 and its training |
| | [AlphaFold3 Explained](https://www.ai4pharm.info/alphafold3)| An explanation of the architecture of Alphafold3 and its training |
| | [AlphaFold 3, Demystified](https://github.com/shenyichong/alphafold3-architecture-walkthrough)| Detailed explanation of the inputs, outputs, and algorithms of AlphaFold 3 |
| ⭐| [The Illustrated AlphaFold](https://elanapearl.github.io/blog/2024/the-illustrated-alphafold/) | A visual walkthrough of the AlphaFold3 architecture with detailed diagrams |
| | [An Opinionated AlphaFold3 Field Guide](https://research.dimensioncap.com/p/an-opinionated-alphafold3-field-guide)| An explanation of the architecture of Alphafold3 in simple-to-understand terms |
| | [A Complete Guide to Protein Folding Prediction with RoseTTAFold](https://alchemybio.substack.com/p/a-complete-guide-to-protein-folding) | Theory and implementation of RoseTTAFold algorithms and architecture |
| | [Guide to undestanding PDB data](https://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/introduction) |  Guide to Understanding PDB Data (Also, check this video for [understanding PDB structure metrics](https://www.youtube.com/watch?v=I64r5DXiLN0&ab_channel=BioinformaticsInsights) as well as [Assessing the Quality of 3D Structures](https://www.rcsb.org/docs/general-help/assessing-the-quality-of-3d-structures)) |
| | [PDB learn](https://pdb101.rcsb.org/learn/exploring-the-structural-biology-of-evolution)| Educational content created by the PDB about structure, function and evolution of proteins |
| | [Comprehensive Overview of Bottom-Up Proteomics using Mass Spectrometry](https://pubs.acs.org/doi/10.1021/acsmeasuresciau.3c00068)| Overview of different proteomics methods |
| | [Preparing for successful protein crystallization experiments](https://journals.iucr.org/f/issues/2025/07/00/oq5003/index.html)|  Strategies to optimize factors that can impact crystallization (Check out the talk on [Youtube](https://www.youtube.com/watch?v=m34RI500OFU&ab_channel=InternationalUnionofCrystallography))|
| | [A primer on molecular dynamics](https://www.owlposting.com/p/a-primer-on-molecular-dynamics) |  Introduction to molecular dynamics |
| | [Anatomy of BoltzGen](https://huggingface.co/spaces/ludocomito/anatomy-of-boltzgen) | Description of BoltzGen architecture (i.e. an AF3-like method) | 
| | [Best practices for presenting and sharing AlphaFold models in a paper](https://australian-structural-biology-computing.github.io/website/best-practices-alphafold) | General guidelines on how to best present AlphaFold model |
| |[A tutorial on predicting various protein-substrate complexes using AlphaFold3 and alphafold3_tools](https://qiita.com/Ag_smith/items/0a2cae3e3125e6895a23)| modeling protein-substrate complexes |
| |[Protein Simulations](https://www.erastova.xyz/teaching/practical-simulations-for-molecules-and-materials/protein-simulations/)|basics of Molecular Dynamics (MD) simulations with GROMACS and VMD, xmgrace, gedit|
| |[Colab notebooks on Biology. Chemistry. Machine Learning.](https://colab-db.github.io/)| 24 Notebooks about molecular analysis |
| |[Structure Analysis and Comparison](https://rbvi.github.io/chimera-tutorials/presentations/modules/chimerax-comp-structures/index.html#/)| Structure Analysis with ChimeraX |
| ⭐ | [Introductory Tutorials for Simulating Protein Dynamics with GROMACS](https://pubs.acs.org/doi/10.1021/acs.jpcb.4c04901) | Simulating a monomer, a complex, and the unfolding of a short polypeptide with GROMACS| 
| ⭐ | [ColabFold protocol](https://github.com/steineggerlab/colabfold-protocol) | predicting protein mono-, multi-mer structures and conformations using ColabFold (check out the [reference paper](https://www.nature.com/articles/s41596-024-01060-5))|
| | [autodock-vina tutorials](https://autodock-vina.readthedocs.io/en/latest/colab_examples.html) | how to run Autodock Vina, and the associated protein and ligand setup utilities on Google Colab | 
| | [Amelie Schreiber Articles](https://huggingface.co/AmelieSchreiber) | use language models and difussion models to analyze proteins | use Generative AI and Protein Language Models to design novel enzymes, predict mutations and binding sites, and analyze protein structures  |
| | [peptidy tutorials](https://github.com/molML/peptidy/tree/main/examples) | classification and generation of antimicrobial peptides xgboost, CNN, LSTM using keras and pytorch | 
| | [Protein Folding with AlphaFold2 Using a Single Sequence](https://colab.research.google.com/drive/1m8mMC4fBg3oSE2PeDjeWGUSqDyCPkf80) | iterative AlphaFold2 predictions to model protein folding pathways and identify potential folding intermediates |
|⭐| [Building Transformer Models for Proteins From Scratch](https://github.com/naity/protein-transformer) | Implement, train, tune, and evaluate a transformer model for antibody classification |
| | [ChimeraX Tutorial](https://dasher.wustl.edu/bio5357/software/chimerax/tutorial.pdf) | ChimeraX basics | 
| | [ChimeraX_overview](https://static-bcrf.biochem.wisc.edu/tutorials/chimerax/Chimera_X_overview_document.html) | ChimeraX overview | 
| | [UCSF ChimeraX Tutorials ChimeraX](https://www.rbvi.ucsf.edu/chimerax/tutorials.html)| visualization, analysis, and animation of molecular structures using ChimeraX  |
| | [ESMBind (ESMB): Low Rank Adaptation of ESM-2 for Protein Binding Site Prediction](https://huggingface.co/blog/AmelieSchreiber/esmbind) | Fine-tuning ESM2 usgin LoRA for Protein Binding Site Prediction|


# Bioinformatics
## Courses
|  | Name | Description |
|-----------|-----------|-----------|
| | [Running and parsing BLAST using the Linux command line](https://vinuesa.github.io/TIB-filoinfo/sesion3_BLAST/) | Basics of the BLAST algorithm and its results, along with useful AWK, Bash, Perl, and R code |
| | [Multiple sequence alignments, and profile hidden Markov models](https://vinuesa.github.io/TIB-filoinfo/sesion4_alineamientos/)| Command-line tutorial on multiple sequence alignment and HMM-based homology search using Clustal and HMMER |
| ⭐| [An Introduction to Applied Bioinformatics](https://readiab.org/introduction.html)| Core concepts in bioinformatics and algorithms with scikit-learn and scikit-bio |
| | [Jupyter Notebook Tutorials for Machine Learning in Chemistry](https://xuhuihuang.github.io/mlchem/html/index.html) | Machine Learning in Chemistry (Check out the [reference paper](https://chemrxiv.org/engage/chemrxiv/article-details/68e029acdfd0d042d1e0312e))| 
|⭐| [Curso avanzado de programación AWK y Bash para bioinformática y biocómputo en sistemas GNU/Linux](https://vinuesa.github.io/intro2linux/index.html) | Introduction to AWK and Bash programming for bioinformatics (in Spanish)|
## Useful publications
| | Name | Description |
|-----------|-----------|-----------|
| | [NEJM Illustrated-glossary](https://illustrated-glossary.nejm.org/)| Glossary for biomedical research |
| | [Glittr](https://glittr.org/) | GitHub repositories with educational material for the computational life sciences |
| ⭐ | [Sandbox BIO](https://sandbox.bio/) | Interactive bioinformatics tutorials |
| | [TomoGuide ](https://tomoguide.github.io/)| Cryo-electron tomography overview adn and a step-by-step tutorial for data processing |
| | [Bio Data Zoo](https://github.com/omgenomics/bio-data-zoo)| Example data in various genomics file formats |
| | [PCR Primer Design](https://binomicalabs.notion.site/PCR-Primer-Design-8f1f888f50a0403a87bdce20e37840f2)| How to design PCR Primers using Primer3 |
| | [Using BLAST on the command line](https://jvanheld.github.io/using_IFB_NNCR/practicals/blast_proteome/blast_proteome.html)| How to search and analyze protein sequnces with BLAST |
| | [LifeMap](https://lifemap.cnrs.fr/)| An interactive tree of life based on NCBI data | 
| | [A-microbiologist-wants-to-compute](https://github.com/SvetlanaUP/A-microbiologist-wants-to-compute) | learning resources related to microbiome computational science and tools |
| | [Bioinformática Microbiana](https://bioinformatica.ciad.mx/home)  | Analysis of microbial genomes (in Spanish) |
| |[Some Thoughts on Splitting Chemical Datasets](https://practicalcheminformatics.blogspot.com/2024/11/some-thoughts-on-splitting-chemical.html)| Splitting Chemical Datasets |


# Machine Learning, Deep Learning, and Data Science
## Courses
|  | Name | Description |
|-----------|-----------|-----------|
| | [DataScienceInteractivePython](https://github.com/GeostatsGuy/DataScienceInteractivePython)| Interactive dashboards for learning data science |
| | [Applied Machine Learning in Python: a Hands-on Guide with Code](https://geostatsguy.github.io/MachineLearningDemos_Book/intro.html) | Introduction to data analysis, machine learning, and neural networks with Python |
| | [Alice’s Adventures in a differentiable wonderlandPermalink](https://www.sscardapane.it/alice-book) | Introduction to differentiable programming and neural network design with PyTorch and JAX |
| | [Introduction to Machine Learning for chemists](https://github.com/ML4chemArg/Intro-to-Machine-Learning-in-Chemistry) | Introduction to Machine Learning for chemists |
| | [A Comprehensive Guide to Explainable AI: From Classical Models to LLMs](https://arxiv.org/abs/2412.00800)| A comprehensive guide to XAI |
| | [Dive into Deep Learning](https://d2l.ai/index.html) | Interactive deep learning book implemented with PyTorch, NumPy/MXNet, JAX, and TensorFlow |
| | [Foundations of Computer Vision](https://visionbook.mit.edu/)| Computer vision, with an image processing and machine learning perspective |
| | [Understanding Deep Learning](https://udlbook.github.io/udlbook/)| Introduction to deep learning |
| ⭐ | [Deep Learning with Python 3th ed](https://deeplearningwithpython.io/)| Introduction to deep learning and main frameworks |
| | [Data Science Fundamentals](https://www.youtube.com/playlist?list=PLOtl7M3yp-DVEdjHo41QS0bLWJi5KMk5m)| data preprocessing, statistical analysis, machine learning, and big data techniques |
| | [Machine Learning For Beginners with python](https://www.youtube.com/playlist?list=PLeo1K3hjS3uvCeTYTeyfe0-rN5r8zn9rw) | 42 tutorials to get started in ML |

## Useful publications
|  | Name | Description |
|-----------|-----------|-----------|
| | [MLU-EXPLAIN](https://mlu-explain.github.io/) | Visual explanations of core machine learning concepts |
| | [Distill publications](https://distill.pub/) | Visual explanations of core machine and deep learning concepts |
| | [Neural Network SandBox](playground.tensorflow.org) | Interactive visualization of neural networks |
| | [ML Code Challenges](https://www.deep-ml.com/) | Practice Machine Learning and Data Science problems |
| | [How to avoid machine learning pitfalls: a guide for academic researchers](https://arxiv.org/abs/2108.02497)| Common mistakes when using machine learning, and how to avoid them |
| ⭐ | [PyTorch in One Hour](https://sebastianraschka.com/teaching/pytorch-1h/)| introduction to PyTorch, from Tensors to Training Neural Networks on Multiple GPUs |
| | [15 Ways to Visualize Regression Results](https://rpubs.com/timothyfraser/tutorial_15_ways_to_visualize_regression_results)| Visualize your regression results|


# Transformers, LLMs, Diffusion Models and Generative AI
## Courses
| | Name | Description |
|-----------|-----------|-----------|
| | [Large Language Model Course](https://github.com/mlabonne/llm-course) | Get into LLMs with roadmaps and Colab notebooks |
| | [Machine Learning Engineering](https://github.com/stas00/ml-engineering) | Training and fine-tuning of LLMs and multi-modal models and their inference | 
| | [Build a Large Language Model (From Scratch)](https://github.com/rasbt/LLMs-from-scratch) | Implement a ChatGPT-like LLM in PyTorch from scratch |
| | [IA generativa](https://microsoft.github.io/generative-ai-for-beginners/#/) | |
| | [How to Scale Your Model](https://jax-ml.github.io/scaling-book/)| Scaling language models |
| | [The Ultra-Scale Playbook](https://huggingface.co/spaces/nanotron/ultrascale-playbook)| How to train LLMs on GPU Clusters (Check out this [video for an overview](https://www.youtube.com/watch?v=1E8GDR8QXKw) as well as this notes [notes](https://damek.github.io/STAT-4830/section/12/notes.html))|
| | [ Flow Matching Guide and Code](https://arxiv.org/abs/2412.06264)| Comprehensive and self-contained review of Flow Matching (i.e. the basis for difussion models)|
| | [Reinforcement Learning: An Overview](https://arxiv.org/abs/2412.05265)| Overview of the field of reinforcement learning and sequential decision making |
| | [The Ultimate Guide to Fine-Tuning LLMs from Basics to Breakthroughs](https://arxiv.org/abs/2408.13296)| Fine-tuning LLMs, from data and optimization to deployment and alignment |
| | [LLM Inference Handbook](https://bentoml.com/llm/)| LLM inference |
| | [The Smol Training Playbook: The Secrets to Building World-Class LLMs](https://huggingface.co/spaces/HuggingFaceTB/smol-training-playbook) | How to design and train a LLM from scratch | 
| | [Stanford CME295 Transformers & LLMs](https://cme295.stanford.edu/syllabus/) | An introduction to transformers, LLMs, reasoning models, and their deployment| 
| ⭐ | [Huggingface Courses](https://huggingface.co/learn) | Learn to use the Huggingface ecosystem (Check out the [lm-course](https://huggingface.co/learn/llm-course/chapter1/1) and [smol-course](https://huggingface.co/learn/smol-course/unit0/1) for LLMs training and fine-tuning as well as the [Diffusion Models Course](https://huggingface.co/learn/diffusion-course/unit0/1)|)|
| | [The Principles of Diffusion Models](https://www.arxiv.org/abs/2510.21890)|  Introduction to diffusion modeling  |
| | [Interpretability of Large Language Models](https://github.com/mega002/llm-interp-tau)| interpretability of LLMs |

## Useful publications
|  | Name | Description |
|-----------|-----------|-----------|
| | [Generative AI exists because of the transformer](https://ig.ft.com/generative-ai/) | A visual explanation of Transformers |
| | [Diffusion Models](https://andrewkchan.dev/posts/diffusion.html) | Notes on the theory behind models like Stable Diffusion and their applications |
| | [Transformer Explainer](https://huggingface.co/papers/2408.04619) | Interactive visualization of Transformers | 
| | [LLM Visualization](https://bbycroft.net/llm) | Interactive visualization of Transformers |
| | [Step-by-Step Diffusion: An Elementary Tutorial](https://arxiv.org/abs/2406.08929)| Diffusion models and flow matching aimed at a technical audience with no diffusion experience |
| | [Guide to Spherical Equivariant Graph Transformers](https://alchemybio.substack.com/p/spherical-equivariant-graph-transformer)| Spherical equivariant GNNs and SE(3)-Transformers explained with protein and molecular examples |
| | [Dummy's Guide to Modern LLM Sampling](https://rentry.org/samplers)| Tokenizers and LLM inference |
| | [LoRA Hyperparameters Guide](https://docs.unsloth.ai/get-started/fine-tuning-guide/lora-hyperparameters-guide)| Best practices for LoRA hyperparameters and how they affect the fine-tuning process |
| | [LLM Embeddings Explained](https://huggingface.co/spaces/hesamation/primer-llm-embedding)| f}Fundamentals of embeddings |
| | [Transformers from Scratch](https://e2eml.school/transformers.html)| Overwiew of the Transformer architecture  |
|| [From Random Forests to RLVR: A Short History of ML/AI Hello Worlds](https://sebastianraschka.com/blog/2025/hello-world-ai.html) | Timeline of ML/DL/GenAI “Hello World” examples, from Random Forests to RL-trained LLMs |

# Python and computer science  
## Courses
|  | Name | Description |
|-----------|-----------|-----------|
| | [cpu.land](https://cpu.land/) | How multiprocessing works, what system calls really are, how computers manage memory|
| | [Fundamentos de ciencia de datos](https://github.com/thatcsharpguy/df) | Introduction to data sceince (in spanish)|
| | [Python for Data Analysis](https://wesmckinney.com/book/)| Data Wrangling with pandas, NumPy, and Jupyter |
| | [Algorithmica](https://en.algorithmica.org/)| Introduction to high performance computing |
| | [The Open-Source Project](https://digital-work-lab.github.io/open-source-project/)| Python package development (Check out the reference [paper](https://github.com/openjournals/jose-reviews/issues/292#issuecomment-3057699562)) |
|⭐| [Diplomado en Ciencia de Datos con Python](https://github.com/GAL-Repository/EDA_Stuff) | Introduction to python and statistical analysis with numpy, seaborn and pandas (in Spanish)| 
|⭐| [MIT 6.100L Introduction to computer science and Programming using Python](https://www.youtube.com/playlist?list=PLUl4u3cNGP62A-ynp6v6-LGBCzeH3VAQB) | Introduction to computer science and Programming | 
| | [Introduction to Bash Scripting](https://github.com/bobbyiliev/introduction-to-bash-scripting/) | Basics of Bash scripting |
| | [Dan Kornas courses](https://www.youtube.com/@dankornas/playlists) | Fundamentals of python, numpy, pandas, matplotlib, machine learning and pytorch |
## Useful publications
| | Name | Description |
|-----------|-----------|-----------|
| | [Developer Roadmaps](https://roadmap.sh/) | Roadmaps to help developers choose a learning path |
| | [Python Tutor](https://pythontutor.com/)| Step-by-step visual debugger and AI tutor to help you understand and debug code |
| | [Introduction to Python](https://carpentries-incubator.github.io/pangenomics-python/) | Introduction to Python |
| | [pandas-cookbook](https://github.com/jvns/pandas-cookbook/) | Recipes for using Python's pandas library |
| | [RegexLearn](https://regexlearn.com/es) | Learn Regex step by step |
| | [GPU Programming: When, Why and How?](https://enccs.github.io/gpu-programming/)| Key concepts in GPU programming |
| | [Bash scripting quirks & safety tips](https://jvns.ca/blog/2017/03/26/bash-quirks/) | Bash basics and best practices for scripting | 
| | [Interactive Map of All Pandas Operations](https://blog.dailydoseofds.com/p/interactive-mind-map-of-all-pandas) | Pandas methods grouped by operation |

# Math
## Courses
|  | Name | Description |
|-----------|-----------|-----------|
| | [Seeing-Theory](https://seeing-theory.brown.edu/) | Introduction to probability and statistics with interactive visualizations |
| | [Mathematical theory of deep learning](https://arxiv.org/abs/2407.18384)| Introduction to the mathematical analysis of deep learning |
| | [Immersive linear algebra](https://immersivemath.com/ila/index.html)| Introduction to linear algebra with interactive visualizations |
| | [Interactive Linear Algebra](https://personal.math.ubc.ca/~tbjw/ila/index.html)| Interactive Linear Algebra |
| ⭐ | [Essence of linear algebra by 3Blue1Brown](https://www.youtube.com/playlist?list=PLZHQObOWTQDPD3MizzM2xVFitgF8hE_ab) | Core concept of linear algebra with a visuals-first approach (Check out the [spanish version of this channel](https://www.youtube.com/playlist?list=PLIb_io8a5NB2DddFf-PwvZDCOUNT1GZoA))|

## Useful publications
|  | Name | Description |
|-----------|-----------|-----------|
| | [Stats illustrations](https://github.com/allisonhorst/stats-illustrations) | R & stats illustrations |
| | [The Matrix Calculus You Need For Deep Learning](https://explained.ai/matrix-calculus/index.html) | Matrix calculus for deep learning, no math background assumed |
| | [Matrix Calculus (for Machine Learning and Beyond)](https://arxiv.org/abs/2501.14787)| Differential calculus to functions on more general vector spaces |


# Data visualization 
| Web | 
|-----------|
| [From Data to Viz](https://www.data-to-viz.com/) |
| [The Python Graph Gallery](https://python-graph-gallery.com/) |
| [Python Color Palette Finder](https://python-graph-gallery.com/color-palette-finder/) |
| [Dataviz caveats](https://www.data-to-viz.com/caveats.html) |
| [FriendsDontLetFriends](https://github.com/cxli233/FriendsDontLetFriends) |
| [r psychologist](https://rpsychologist.com/viz) |

# Blogs 
| Web | Topic |
|-----------|-----------|
| [GAMA Miguel Angel (My blog)](https://miangoaren.github.io/year-archive/) | Protein science | 
| [Diego del Alamo](https://publish.obsidian.md/ddelalamo/) | Protein science |
| [Liam Bai](https://liambai.com/) | Protein science |
| [Owl Posting](https://www.owlposting.com/) | Protein science |
| [Epoch AI](https://epochai.org/blog) | AI (Check out the list of [Bio models](https://epoch.ai/blog/biological-sequence-models-in-the-context-of-the-ai-directives))|
| [Lilian Weng](https://lilianweng.github.io/) | AI |
| [Practical Cheminformatics](https://practicalcheminformatics.blogspot.com/) | Chemistry (for more [resources](https://github.com/PatWalters/resources_2025)) |
| [Oxford Protein Informatics Group](https://www.blopig.com/blog/) | Protein science |
| [James Lingford](https://www.jameslingford.com/blog/) | Bioinformatics |
| [Boolean Biotech](https://blog.booleanbiotech.com/archives) | Protein science |
| [XGblog](https://www.xgblog.ai/) | AI |

