# For protein sequences
| Name | Description | 
|-----------|-----------| 
| [SeqKit](https://bioinf.shenwei.me/seqkit/) ([tutorial](https://sandbox.bio/tutorials/seqkit-intro)) | ultrafast toolkit for FASTA/Q file manipulation |
| [hh-suite](https://bioinf.shenwei.me/seqkit/) | remote protein homology detection suite |
| [Diamond2](https://github.com/bbuchfink/diamond) | Accelerated BLAST compatible local sequence aligner |
| [MMseqs2](https://github.com/soedinglab/MMseqs2) | ultra fast and sensitive search and clustering suite (see also the [gpu-support](https://github.com/soedinglab/MMseqs2/wiki#compile-from-source-for-linux-with-gpu-support)) |
| [ProDy](http://prody.csb.pitt.edu/tutorials/) | protein structure, dynamics, and sequence analysis |
| [seqlike](https://github.com/modernatx/seqlike) |  Unified biological sequence manipulation |
| [BioNumpy](https://github.com/bionumpy/bionumpy/) | array programming on biological datasets |
| [pLM-BLAST](https://github.com/labstructbioinf/pLM-BLAST) | detection of remote homology by protein language models |  
| [PLMSearch](https://github.com/maovshao/PLMSearch) | homologous protein search with protein language models |
| [iSeq](https://github.com/BioOmics/iSeq) | download data from sequence databases like GSA, SRA, ENA, and DDBJ |
| [LexicMap](https://github.com/shenwei356/LexicMap) | sequence alignment against millions of genomes |
| [NCBI datasets](https://www.ncbi.nlm.nih.gov/datasets/docs/v2/download-and-install/) | download data from NCBI databases |
| [ncbi-genome-download](https://github.com/kblin/ncbi-genome-download) | scripts to download genomes from the NCBI FTP servers |
| [ncbi-acc-download](https://github.com/kblin/ncbi-acc-download) | download files from NCBI Entrez by accession |
| [ProtNLM](https://www.uniprot.org/help/ProtNLM) | UniProt's Automatic Annotation pipeline for protein sequences (see the [Colab notebook](https://colab.research.google.com/github/google-research/google-research/blob/master/protnlm/protnlm_evidencer_uniprot_2023_01.ipynb)) |
| []() |  |


# For protein structures
| Name | Description | 
|-----------|-----------| 
| [BioPandas](https://biopandas.github.io/biopandas/) | working with molecular structures in pandas |
| [FoldSeek](https://github.com/steineggerlab/foldseek) | fast and sensitive comparisons of large structure sets|
| [foldcomp](https://github.com/steineggerlab/foldcomp) | Compressing protein structures |
| [Foldmason](https://github.com/steineggerlab/foldmason) | multiple Protein Structure Alignment at Scale |
| [folddisco](https://github.com/steineggerlab/folddisco) | indexing and search of discontinuous motifs |
| [PyPDB](https://github.com/williamgilpin/pypdb) | python API for the PDB |
| [afpdb](https://github.com/data2code/afpdb) | efficient manipulation of protein structures in Python |
| [LocalPDB](https://github.com/labstructbioinf/localpdb) | manage protein structures and their annotations |
| [pdb-tools](https://github.com/haddocking/pdb-tools) | manipulating and editing PDB files |
| [pdbfixer](https://github.com/openmm/pdbfixer) | fixes problems in PDB files |
| [PDBe API Training Notebooks](https://github.com/glevans/pdbe-api-training) | for understanding how the PDBe REST API works |
| [USalign](https://github.com/pylelab/USalign) |structure alignment of monomeric and complex proteins and nucleic acids |
| [cath-tools](https://github.com/UCLOrengoGroup/cath-tools) | structure comparison tools |
| [Pyrcsbsearchapi](https://github.com/rcsb/py-rcsbsearchapi) | Python interface for the RCSB PDB search API |
| [protestar](https://github.com/refresh-bio/protestar) | compress collections structures |
| [Merizo-search](https://github.com/psipred/merizo_search) | domain segmentation |
| [progres](https://github.com/greener-group/progres) | structure searching by structural embeddings |
| [freesasa](https://github.com/mittinatten/freesasa) | for calculating Solvent Accessible Surface Areas |
| [openstructure](https://openstructure.org/docs/2.8/) | protein structure, complexes and docking comparison |
| [opendock](https://github.com/guyuehuo/opendock)| protein-Ligand Docking and Modeling |
| [pyScoMotif](https://github.com/3BioCompBio/pyScoMotif) | protein motif search |
| [pyRMSD](https://github.com/salilab/pyRMSD) | RMSD calculations of large sets of structures |
| [Muscle-3D](https://github.com/rcedgar/muscle) | multiple protein structure alignment |
| [pdb-redo](https://pdb-redo.eu/)| automated procedure to refine, rebuild and validate your model |
| [proteinshake](https://github.com/BorgwardtLab/proteinshake) | preprocessed and cleaned protein 3D structure datasets |
| [profet](https://github.com/alan-turing-institute/profet) | Retrieves the cif or pdb files from either thePDB (using pypdb) or Alphafold using the Uniprot ID |
| [mini3di](https://github.com/althonos/mini3di) | NumPy port of the foldseek code for encoding protein structures to 3di |
| []() |  |

# Representation learning for proteins
| Name | Description | 
|-----------|-----------|
| [ProtLearn](https://github.com/tadorfer/protlearn) | extracting protein sequence features |
| [Pfeature](https://github.com/raghavagps/Pfeature) | computing features of peptides and proteins |
| [bio_embeddings](https://github.com/sacdallago/bio_embeddings) | compute protein embeddings from sequences |
| [Graph-Part](https://github.com/graph-part/graph-part) | data partitioning method for ML |
| [ProteinFlow](https://github.com/adaptyvbio/ProteinFlow) | data processing pipeline fo ML |
| [docktgrid](https://github.com/gmmsb-lncc/docktgrid) | Create customized voxel representations of protein-ligand complexes |
| [Prop3D](https://github.com/bouralab/Prop3D) | toolkit for protein structure dataset creation and processing  |
| [SaProt](https://github.com/westlake-repl/SaProt) | Protein Language Model with Structural Alphabet (AA+3Di) |
| [ProstT5](https://github.com/mheinzinger/ProstT5) | Bilingual Language Model for Protein Sequence and Structure (see the [Foldseek adaptation](https://github.com/steineggerlab/foldseek?tab=readme-ov-file#structure-search-from-fasta-input)) |
| [Graphein](https://github.com/a-r-j/graphein) | geometric representations of biomolecules and interaction networks |
| [PyUUL](https://pyuul.readthedocs.io/index.html) | encode structures into differentiable data structures |
| [colav](https://github.com/Hekstra-Lab/colav) | feature extraction methods like dihedral angles, CA pairwise distances, and strain analysis |
| [ProTrek](https://github.com/westlake-repl/ProTrek) <br> [webserver](http://search-protrek.com/)| multimodal (sequence-structure-function) protein representations and annotations |
| [masif](https://github.com/LPDI-EPFL/masif) | molecular surface interaction fingerprints |
| [xxx](xxx) | xxx |



# Protein engineering
| Name | Description | 
|-----------|-----------|
| [biotite](https://www.biotite-python.org/latest/) | sequence and structure manipulation and analysis |
| [protkit](https://github.com/silicogenesis/protkit) | Unified Approach to Protein Engineering |
| [ProteinMPNN in JAX](https://github.com/sokrypton/ColabDesign/tree/main/mpnn) | Fast implementation of ProteinMPNN |
| [EvoProtGrad](https://github.com/NREL/EvoProtGrad) | directed evolution with MCMC and protein language models |
| [ConservFold](https://www.rodrigueslab.com/resources) | map amino acid conservation intro structures with the AF2 pipeline |
| [consurf](https://consurf.tau.ac.il/consurf_index.php) |identification of functionally important regions in proteins |
| [AlphaPulldown](https://www.embl-hamburg.de/AlphaPulldown/) | Complex moedeling with AF-Multimer |
| [ColabDock](https://github.com/JeffSHF/) | protein-protein docking |
| [ColabDesign](https://github.com/sokrypton/ColabDesign) | protein design pipelines |
| [LazyAF](https://github.com/ThomasCMcLean/LazyAF) | protein-protein interaction with AF2|
| [CombFold](https://github.com/dina-lab3D/CombFold) | structure predictions of large complexes |
| [Cfold](https://github.com/patrickbryant1/Cfold) | structure prediction of alternative protein conformations |
| [Replacement Scan](https://colab.research.google.com/github/sokrypton/ColabBio/blob/main/notebooks/replacement_scan.ipynb) | find how many amino acid replacements your protein can tolerate [see tw](https://x.com/sokrypton/status/1812769477228200086) |
| [protein_scoring](https://github.com/seanrjohnson/protein_scoring) | generating and scoring novel enzyme sequences  |
| [AF_unmasked](https://github.com/clami66/AF_unmasked) | structure prediction for huge protein complexes (~27 chains and ~8400aa) |
| [AncFlow](https://github.com/rrouz/AncFlow) | pipeline for the ancestral sequence reconstruction of clustered phylogenetic subtrees |
| [TRILL](https://github.com/martinez-zacharya/TRILL) | Sandbox for Deep-Learning based Computational Protein Design |
| [AF2BIND](https://github.com/sokrypton/af2bind) | Predicting ligand-binding sites based on AF2 |
| [ProteinMPNN](https://github.com/dauparas/ProteinMPNN) | Fixed backbone design ([see webserver](https://huggingface.co/spaces/simonduerr/ProteinMPNN))|
| [ligandMPNN](https://github.com/dauparas/LigandMPNN) | Fixed backbone design sensible to ligands ([see colab notebook](https://github.com/ullahsamee/ligandMPNN_Colab))|
| [PyPEF](https://github.com/Protein-Engineering-Framework/PyPEF) | sequence-based machine learning-assisted protein engineering |
| [DeepProtein](https://github.com/jiaqingxie/DeepProtein) | protein Property Prediction |
| [FlexMol](https://github.com/Steven51516/FlexMol) | construction and evaluation of diverse model architectures  |
| []() |  |




# For molecules
| Name | Description | 
|-----------|-----------| 
| [PDBe CCDUtils](https://pdbeurope.github.io/ccdutils/index.html)  | tools to deal with PDB chemical components and visualization ([see also](https://github.com/PDBeurope/pdbe-notebooks/tree/main/pdbe_ligands_tutorials))|
| [PDBe Arpeggio](https://github.com/PDBeurope/arpeggio) |  calculation of interatomic interactions in molecular structures|
| [PDBe RelLig](https://github.com/PDBeurope/rellig) | classifies ligands based on their functional role| 
| [MolPipeline](https://github.com/basf/MolPipeline) | processing molecules with RDKit in scikit-learn |
| [roshambo](https://github.com/molecularinformatics/roshambo) | molecular shape comparison |
| [FlexMol](https://github.com/Steven51516/FlexMol) | construction and evaluation of diverse model architectures |
| [molli](https://github.com/SEDenmarkLab/molli) | general purpose molecule library generation and handling  |
| [useful_rdkit_utils](https://github.com/PatWalters/useful_rdkit_utils) | A collection of useful RDKit and sci-kit learn functions |
| []() |  |


# For machine learning
| Name | Description | 
|-----------|-----------| 
| [Colab forms](https://colab.research.google.com/notebooks/forms.ipynb) | how to convert a colab notebook to a user interface |
| [cuml](https://github.com/rapidsai/cuml) | GPU-based implementations of common machine learning algorithms ([more info for umap optimization](https://developer.nvidia.com/blog/even-faster-and-more-scalable-umap-on-the-gpu-with-rapids-cuml/))|
| [LazyPredict](https://github.com/shankarpandala/lazypredict) | build a lot of basic models without much code |
| [TorchDR](https://github.com/TorchDR/TorchDR) | PyTorch Dimensionality Reduction |
| [Kerasify](https://github.com/moof2k/kerasify) | running trained Keras models from a C++ application |
| [pca](https://erdogant.github.io/pca/pages/html/index.html) | perform PCA and create insightful plots |
| [openTSNE](https://opentsne.readthedocs.io/en/stable/) | faster implementation of t-SNE that includes other optimizations |
| []() |  |


# statistics
| Name | Description | 
|-----------|-----------| 
| [scikit-posthocs](https://scikit-posthocs.readthedocs.io/en/latest/) |  post hoc tests for pairwise multiple comparisons |
| [statannotations](https://github.com/trevismd/statannotations) | add statistical significance annotations on seaborn plots |
| [ggstatsplot](https://github.com/IndrajeetPatil/ggstatsplot) | creating graphics with details from statistical tests included in the information-rich plots themselves |
| [statsmodels](https://www.statsmodels.org/stable/index.html) | classes and functions for the estimation of many different statistical models, as well as for conducting statistical tests, and statistical data exploration |
| []() |  |


# Data viz for bio
| Name | Description | 
|-----------|-----------| 
| [MolecularNodes](https://github.com/BradyAJohnston/MolecularNodes) | Toolbox for molecular animations in Blender |
| [CellScape](https://github.com/jordisr/cellscape) | Vector graphics cartoons from protein structure |
| [ChimeraX apps](https://cxtoolshed.rbvi.ucsf.edu/apps/all) | chimeraX  extensions |
| []() |  |


# Webservers 
| Web | Description | 
|-----------|-----------| 
| [ProteInfer](https://google-research.github.io/proteinfer/) | predicting functional properties from sequences |
| [GoPredSim](https://embed.protein.properties/) | Predict protein properties from embeddings |
| [DeepFRI](https://beta.deepfri.flatironinstitute.org/) | structure-based protein function prediction and functional residue identification |
| [protein structure relaxation](https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/beta/relax_amber.ipynb) | Relax your structure using OpenMM/Amber |
| [Model Memory Calculator](https://huggingface.co/spaces/hf-accelerate/model-memory-usage) | calculate how much vRAM is needed to train and perform inference on a model hosted on Hugging Face (see also [gpu_poor calculator](https://github.com/RahulSChand/gpu_poor)) |
| [alphafind](https://alphafind.fi.muni.cz/search) | structure-based search |
| [DiffDock-Web](https://huggingface.co/spaces/reginabarzilaygroup/DiffDock-Web) | molecular docking with ligands |
| [ESMFold](https://esmatlas.com/resources?action=fold) | structure prediction with ESMFold |
| [Foldseek clusters](https://cluster.foldseek.com/) | search for sctructural clusters in AFDB |
| [damietta](https://damietta.de/) | protein design toolkit |
| [easifa](http://easifa.iddd.group/) | active and binding site annotations for enzymes |
| [InterProt](https://interprot.com/#/) |  interpretability of features derived from protein language models using sparse autoencoders |
| [MPI Bioinformatics Toolkit!](https://toolkit.tuebingen.mpg.de/) | multiple bioinformatics tools |
| [moleculatio](https://moleculatio.yamlab.app) | chemoinformatics, quantum chemistry and molecular dynamics simulations or small molecules |
| []() |  |
 
# Data Viz
| Name | Description | 
|-----------|-----------| 
| [datamapplot](https://github.com/TutteInstitute/datamapplot) | creating beautiful, interactive and massive scatterplots |
| []() |  |


# ChatBots 
| Tema | 
|-----------|
| [ChatGPT](https://chat.openai.com/) |
| [Gemini](https://gemini.google.com/) |
| [claude](https://claude.ai/) |
| [Bing](https://www.bing.com/?cc=es) |
| [HuggingChat](https://huggingface.co/chat/) |
| [biologpt](https://biologpt.com/) |
| [consensus](https://consensus.app/) |  
| [typeset](https://typeset.io/) |  
| [mistral-chat](https://mistral.ai/news/mistral-chat/) |  
| []() | 

