# For sequences
| Name | Descripcion | 
|-----------|-----------| 
| [SeqKit](https://bioinf.shenwei.me/seqkit/) & [short tutorial](https://sandbox.bio/tutorials/seqkit-intro)| Manipulacion de secuencias genomicas |
| [hh-suite](https://bioinf.shenwei.me/seqkit/) | Busqueda de secuencias homólogas remotas  |
| [Diamond2](https://github.com/bbuchfink/diamond) | Blasteo de secuencias de proteinas a escala masiva |
| [MMseqs2](https://github.com/soedinglab/MMseqs2) | Blasteo de secuencias de proteinas a escala masiva |
| [FoldSeek](https://github.com/steineggerlab/foldseek) | Blasteo estructural de proteinas a escala masiva|
| [ProtLearn ](https://github.com/tadorfer/protlearn) | Codificacion de secuencias de proteinas en vectores para el entrenamiento de algoritmos de machine learning |
| [Pfeature](https://github.com/raghavagps/Pfeature) |Codificacion de secuencias de proteinas en vectores para el entrenamiento de algoritmos de machine learning  |
| [Graphein](https://github.com/a-r-j/graphein) | Codificacion de estructuras de proteinas en vectores usando  teoria de grafos |
| [PyUUL](https://pyuul.readthedocs.io/index.html) | Codificacion de estructuras de proteinas en vectores usando algoritmos de analisis 3D |
| [bio_embeddings](https://github.com/sacdallago/bio_embeddings) | Implementacion de varios lenguajes de proteinas  |
| [TRILL ](https://github.com/martinez-zacharya/TRILL) | Implementacion de varios lenguajes de proteinas  |
| [Graph-Part](https://github.com/graph-part/graph-part) |  Preparacion de datasets de secuencias de proteinas  |
| [LazyPredict](https://github.com/shankarpandala/lazypredict) | Comparacion automatica de varios algoritmos de clasificacion y regresion |
| [MolecularNodes](https://github.com/BradyAJohnston/MolecularNodes) | Visualizacion de proteinas con calidad profesional usando Blender |
| [ProDy](http://prody.csb.pitt.edu/tutorials/) | Suite de herramientas de analisis de proteinas |
| [PyPDB](https://github.com/williamgilpin/pypdb) | API para interactuar con la PDB con python |
| [LocalPDB](https://github.com/labstructbioinf/localpdb) | API para interactuar con la PDB con python |
| [pdb-tools](https://github.com/haddocking/pdb-tools) | Herramientas para procesamiento y analisis de archivos .pdb |
| [seqlike](https://github.com/modernatx/seqlike) | Manipulacion y representacion de secuencias de proteinas |
| [BioNumpy](https://github.com/bionumpy/bionumpy/) | Manipulacion y representacion de secuencias usando numpy |
| [pdbfixer](https://github.com/openmm/pdbfixer) | Correccion de errores en archivos .pdb|
| [ProteinFlow](https://github.com/adaptyvbio/ProteinFlow) | procesamiento de secuencias y estructruas para tareas de machine learning|
| [cath DB](https://www.cathdb.info/version/v4_3_0/superfamily/3.40.710.10) | Base de datos de plegamientos proteicos. Ejemplo con la superfamilia de serinbetalactamasas|
| [proteinshake](https://github.com/BorgwardtLab/proteinshake) | procesamiento de estructruas para tareas de machine learning|
| [EvoProtGrad](https://github.com/NREL/EvoProtGrad) | Simular experimentos de evolucion dirigida usando modelos de lenguaje de proteinas|
| [ConservFold](https://www.rodrigueslab.com/resources) | Mapear los residuos conervados a una estructrua usando AlphaFold2 |
| [docktgrid](https://github.com/gmmsb-lncc/docktgrid) | Codificacion de estructuras de proteinas y sus ligandos en vectores usando algoritmos de analisis 3D |
| [pdbeccdutils](https://pdbeurope.github.io/ccdutils/index.html) y [PDBe Arpeggio](https://github.com/PDBeurope/arpeggio) | Manipulacion de pequeñas moluclas de la PDB asi como su interaccion con proteinas |
| [PDBminer](https://github.com/ELELAB/PDBminer) | Manipulacion modelos de la PDB y AlphaFold Database  |
| [pdbe-api-training](https://github.com/glevans/pdbe-api-training) | Tutoriales para aprender a usar la ePDB  |
| [Prop3D](https://github.com/bouralab/Prop3D) | Herramientas para la preparacion de datasets de proteinas para machine learning  |
| [ProteinMPNN in JAX](https://github.com/sokrypton/ColabDesign/tree/main/mpnn) | Una implementacion optimizada de ProteinMPNN en JAX  |
| [BioPandas](https://biopandas.github.io/biopandas/) | Una implementacion de pandas para manipular archivos .pdb  |
| [USalign](https://github.com/pylelab/USalign) | Alineamiento de estructruas monomericas y heterodimeros o en combinacion con acidos nucleicos  |
| [protkit](https://github.com/silicogenesis/protkit) | Manipulacion y representacion de proteinas |
| [foldcomp](https://github.com/steineggerlab/foldcomp) | Algoritmo de compresion de estructruas de proteinas y bases de datos masivas |
| [cath-tools](https://github.com/UCLOrengoGroup/cath-tools) |  Manipulacion y calculos con estructruas proteicas |
| [Pyrcsbsearchapi](https://github.com/rcsb/py-rcsbsearchapi) |  API de la PDB |
| [protestar](https://github.com/refresh-bio/protestar) | Algoritmo de compresion de estructruas de proteinas |
| [consurf](https://consurf.tau.ac.il/consurf_index.php) | Identification de regiones conervadas de proteinas |
| [colav](https://github.com/Hekstra-Lab/colav) | analisis de ensambles conformacionales |
| [AncFlow](https://github.com/rrouz/AncFlow) | pipeline para reconstruccion ancestral |
| [Merizo-search](https://github.com/psipred/merizo_search) | domain segmentation |
| [Foldmason](https://github.com/steineggerlab/foldmason) | Multiple Protein Structure Alignment at Scale |
| [AlphaPulldown](https://www.embl-hamburg.de/AlphaPulldown/) | Complex moedeling with AF-Multimer |
| [ColabDock](https://github.com/JeffSHF/) | Docking protein-proteina |
| [ColabDesign](https://github.com/sokrypton/ColabDesign) | inverse ffolding |
| [LazyAF](https://github.com/ThomasCMcLean/LazyAF) | protein-protein interaction with AF2|
| [progres](https://github.com/greener-group/progres) | Fast protein structure searching by structural embeddings |
| [pLM-BLAST](https://github.com/labstructbioinf/pLM-BLAST) | Remote homology by PLM |
| [CombFold](https://github.com/dina-lab3D/CombFold) | structure of large protein complexes |
| [ChimeraX apps](https://cxtoolshed.rbvi.ucsf.edu/apps/all) | chimeraX  extensions |
| [masif](https://github.com/LPDI-EPFL/masif) | protein desing based on functionl surface patches |
| [Cfold](https://github.com/patrickbryant1/Cfold) | Structure prediction of alternative protein conformations |
| ["Alanin-scanning" with Alphafold2](https://colab.research.google.com/github/sokrypton/ColabBio/blob/main/notebooks/replacement_scan.ipynb) | Structure prediction with a fraction of the aa replaced with Ala [Twitter](https://x.com/sokrypton/status/1812769477228200086) |
| [ColabProTrek](https://colab.research.google.com/drive/1On2xQU0d7351bIBgZpz2T0VUp2gZium0?usp=sharing) and [huggingface server](https://huggingface.co/spaces/westlake-repl/Demo_ProTrek_650M_UniRef50)| Multimodal protein annotation |
| [MolPipeline](https://github.com/basf/MolPipeline) | processing molecules with RDKit in scikit-learn |
| [Prop3D](https://github.com/bouralab/Prop3D) | toolkit for protein structure dataset creation and processing |
| [Colab forms](https://colab.research.google.com/notebooks/forms.ipynb) | convert a colab notebook to an UI |
| [freesasa](https://github.com/mittinatten/freesasa) | for calculating Solvent Accessible Surface Areas |
| [protein_scoring](https://github.com/seanrjohnson/protein_scoring) | Generating and scoring novel enzyme sequences with a variety of models and metrics |
| [roshambo](https://github.com/molecularinformatics/roshambo) | molecular shape comparison |
| [openstructure](https://openstructure.org/docs/2.8/) | protein structure , complexes and docking comparison |
| [AF_unmasked](https://github.com/clami66/AF_unmasked) | structure prediction for huge protein complexes (up to ~27 chains and ~8400 amino acids) |
| [opendock](https://github.com/guyuehuo/opendock)| Protein-Ligand Docking and Modeling |
| [pyScoMotif](https://github.com/3BioCompBio/pyScoMotif) | protein motif search |
| [folddisco](https://github.com/steineggerlab/folddisco) | protein motif search |
| [FlexMol](https://github.com/Steven51516/FlexMol) | construction and evaluation of diverse model architectures |
| [pyRMSD](https://github.com/salilab/pyRMSD) | RMSD calculations of large sets of structures |
| [PLMSearch](https://github.com/maovshao/PLMSearch) | accurate and fast homologous protein search with PLMs |
| [iSeq](https://github.com/BioOmics/iSeq) | Download (meta)data from GSA, SRA, ENA, and DDBJ |
| [SaProt](https://github.com/westlake-repl/SaProt) | Protein Language Model with Structural Alphabet (AA+3Di) |
| [ProstT5](https://github.com/mheinzinger/ProstT5) and [Foldseek adaptation](https://github.com/steineggerlab/foldseek?tab=readme-ov-file#structure-search-from-fasta-input)| Bilingual Language Model for Protein Sequence and Structure |
| [Muscle-3D](https://github.com/rcedgar/muscle?tab=readme-ov-file#multiple-structure-alignment) | scalable multiple protein structure alignment |
| [molli](https://github.com/SEDenmarkLab/molli) | general purpose molecule library generation and handling  |
| [pdb-redo.eu](https://pdb-redo.eu/) & [video explanation](https://www.youtube.com/watch?v=CqgR_HgqRWY&ab_channel=CCP4)| automated procedure to refine, rebuild and validate your model |
| [xxx](xxx) | xxx |
| [xxx](xxx) | xxx |
| [xxx](xxx) | xxx |
| [xxx](xxx) | xxx |
| [xxx](xxx) | xxx |


# For protein structures
| Name | Descripcion | 
|-----------|-----------| 
| [xxx](xxx) | xxx |


# For chemical molecules
| Name | Descripcion | 
|-----------|-----------| 
| [xxx](xxx) | xxx |



# Webservers 
| Web | Descripcion | 
|-----------|-----------| 
| [ProteInfer](https://google-research.github.io/proteinfer/) | predicting the functional properties of a protein from its amino acid sequence using neural networks |
| [GoPredSim](https://embed.protein.properties/) | Predict protein properties from embeddings |
| [ProteinMPNN](https://huggingface.co/spaces/simonduerr/ProteinMPNN) | Fixed backbone design |
| [DeepFRI](https://beta.deepfri.flatironinstitute.org/) | structure-based protein function prediction (and functional residue identification) method using Graph Convolutional Networks with Language Model features |
| [Amber relaxation](https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/beta/relax_amber.ipynb) | Usa AMBER para relajar tus estructruas de proteinas y evitar clashes estericos|
| [Model Memory Calculator](https://huggingface.co/spaces/hf-accelerate/model-memory-usage) | Calcula el peso de cualquier modelo almacenado en HuggingFace|
| [alphafind](https://alphafind.fi.muni.cz/search) | busqueda de estructruas en la AlphaFold database|
| [af2bind](https://colab.research.google.com/github/sokrypton/af2bind/blob/main/af2bind.ipynb) | Prediccion de sitios de union a sustratos pequeños |
| [DiffDock-Web](https://huggingface.co/spaces/reginabarzilaygroup/DiffDock-Web) | Docking de proteinas con ligandos |
| [ligandMPNN](https://colab.research.google.com/github/ullahsamee/ligandMPNN_Colab/blob/main/LigandMPNN_Colab.ipynb) | Notebook para usar LigandMPNN en google colab|
| [ESMFold](https://esmatlas.com/resources?action=fold) | Weberver para plegar proteins con ESMFold |
| [Foldseek clusters](https://cluster.foldseek.com/) | Weberver para identificar clusters de estructuras en la AFDB |
| [damietta](https://damietta.de/) | protein design toolkit |
| [Model Memory Calculator](https://huggingface.co/spaces/hf-accelerate/model-memory-usage) and [gpu_poor calculator](https://github.com/RahulSChand/gpu_poor)| calculate how much vRAM is needed to train and perform inference on a model hosted Hugging Face |
| [AlphaFold Clusters](https://cluster.foldseek.com/) | investigate the Foldseek clustered AlphaFold database |
| [protrek server](http://search-protrek.com/) | search proteins by seq, structure and funtional annotations |
| [easifa](http://easifa.iddd.group/) | active and binding site annotations for enzymes |


# ChatBots de ayuda
| Tema | 
|-----------|
| [ChatGPT](https://chat.openai.com/) |
| [Gemini](https://gemini.google.com/) |
| [claude](https://claude.ai/) |
| [Bing](https://www.bing.com/?cc=es) |
| [HuggingChat](https://huggingface.co/chat/) |
| [biologpt](https://biologpt.com/) |
